spec/requests/api/logging_spec.rb

```diff
@@ -0,0 +1,51 @@
+#
+# REST API Logging Tests
+#
+describe ApiController do
+  describe "Successful Requests logging" do
+    LOG_REQUESTS = {
+      "API Request"    => nil,
+      "Authentication" => nil,
+      "Authorization"  => nil,
+      "Request"        => nil,
+      "Parameters"     => nil,
+      "Response"       => nil
+    }.freeze
+
+    def expect_log_requests(expectations)
+      expectations.each do |category, expectation|
+        expect_any_instance_of(ApiController).to receive(:log_request)
```

Comment:
I'm wondering if the former would still be better? IMHO it would be less coupled to the implementation. For instance, this test is likely to break as we move towards a more OO model for the API code, even though the behavior may still be correct.